src/core/georeferencing.cpp

```diff
@@ -859,6 +859,12 @@ void Georeferencing::setScaleFactors(double combined_scale_factor, double auxili
 	}
 }
 
+double Georeferencing::getDeclinationSetting() const
+{
+	constexpr auto precision_ratio = power<grivationPrecision() - declinationPrecision()>(10);
+	return ((grivation + convergence)*precision_ratio + declination)/(precision_ratio + 1);
```

Comment:
A remark. I don't know the required math skills level to understand this code, and  I spent almost an hour trying to understand how the above calculation works. I may be that I'm not educated enough to get the meaning, or I'm looking at it from the wrong side.
1) I never studied the internals of `Georeferencing`. As to my understanding, cartographers use positive declination values for a shift to the right when looking towards the north. Same for meridian convergence - positive right, negative is left. In my area, declination is about 4°, and meridian convergence of the popular local projection is about 7°. Grivation is a sum of the two numbers, which is 4°+7°=11° and it's aligned with the numbers produced by [the tool](https://mapy.orientacnisporty.cz/en/convergenceclass ) we are using in my country. However, `Georeferencing` uses the opposite sign for convergence. That means ` grivation + convergence` is (counterintuitively) declination. When "positive means rotation to the right - always" would hold true, the declination calculation would be `grivation - convergence`. In other words, it took me some tens of minutes to understand that the calculation in `getDeclinationSetting()` is, in fact, "(declination*1000000+declination)/1000001".
2)  I don't get why the precision ratio is 10^6 and why the divisor contains "+1".
3) The "setting" in the method name associates with some "mode" or "flags" for me. I don't expect a method with such a name to return a number with enhanced precision or some combined figure.

If you decide to address my remarks, please do so through the code and commit message rather than comments in this pull request. Think about future contributors who will fix and enhance this code. In my opinion, they deserve to understand it without excessive effort.

Reply:
How to combine convergence with grivation is one of those conventions that could have gone either way, and intuition doesn't work for me, either. I'm often referring to the code in `setDeclination()` to check whether to add or subtract. An explanation of Mapper's convention can be found via the manual's **Georeferencing: Further Reading** section which refers to [Larry Simons (2005): Magnetic Declination & Grid Convergence](http://www.threelittlemaids.co.uk/magdec/explain.html).

Reply:
@lpechacek Thanks for your helpful remarks after the type of effort you put into it. I plan to address your 3 remarks in code changes (and comments in the code). The alternative name which comes to mind is `getPreciseDeclination()`. I hope that will make sense to our developers.

Reply:
Thanks, Scott, for the pointer. It is far more understandable than anything I read before. I spent a few hours today studying the peculiarities of the national grid and the tool I referenced before. It turned out that the national grid has negative convergence over all the area. However, the tool shows positive converegence figure which drove me into the weeds. It means that, point 1) of my comment is invalid and I'll mark it as such. I appreciate your patience. :)

Reply:
As a side note, my research resulted in the discovery of the following document: [_OGP Geomatics guidance note 21 - Grid convergence_](https://ge0mlib.ru/papers/Guide/IOGP/373-21.pdf). The document mentions (defines?) two different conventions for the meridian convergence figure. One is called "Gauss-Bomford convention" and the other is "Survey convention". They differ in the sign and countries which use them. We may want to document that Mapper is using the Gauss-Bomford convention like [ArcGIS did](https://developers.arcgis.com/net/uwp/api-reference/html/M_Esri_ArcGISRuntime_Geometry_SpatialReference_GetConvergenceAngle.htm).